clinica/pydra/t1_volume/t1_volume_utils.py

```diff
@@ -0,0 +1,536 @@
+    the volume did not exceed 100 mm. Above this distance, either the volume is either not segmented (SPM error), or the
+    produced segmentation is wrong (not the shape of a brain anymore)
+    """
+
+    import numpy as np
+
+    center = get_world_coordinate_of_center(nii_volume)
+
+    distance_from_origin = np.linalg.norm(center, ord=2)
+
+    if distance_from_origin < threshold_l2:
+        return True
+    else:
+        # If center is a np.nan,
+        return False
+
+
+def get_world_coordinate_of_center(nii_volume: PathLike) -> ndarray:
+    """Extract the world coordinates of the center of the image.
+
+    Parameters
+    ---------
+    nii_volume : PathLike
+        path to nii volume
+
+    Returns
+    -------
+    tuple
+        coordinates in the world space
+
+    References
+    ------
+    https://brainder.org/2012/09/23/the-nifti-file-format/
+
+    """
+
+    from os.path import isfile
+
+    import nibabel as nib
+    import numpy as np
+    from nibabel.filebasedimages import ImageFileError
+
+    # from clinica.utils.stream import cprint
+    # assert isinstance(nii_volume, str), "input argument nii_volume must be a str"
+
+    assert isfile(nii_volume), "input argument must be a path to a file"
+
+    try:
+        orig_nifti = nib.load(str(nii_volume))
+    except ImageFileError:
+        print(f"File {nii_volume} could not be read by nibabel. Is it a valid NIfTI")
+        return np.nan
+
+    head = orig_nifti.header
+
+    if isinstance(head, nib.freesurfer.mghformat.MGHHeader):
+        # If MGH volume
+        center_coordinates_world = vox_to_world_space_method_3_bis(
+            head["dims"][0:3] / 2, head
+        )
+    else:
+        # Standard NIfTI volume
+        center_coordinates = get_center_volume(head)
+
+        if head["qform_code"] > 0:
+            center_coordinates_world = vox_to_world_space_method_2(
+                center_coordinates, head
+            )
+        elif head["sform_code"] > 0:
+            center_coordinates_world = vox_to_world_space_method_3(
+                center_coordinates, head
+            )
+        elif head["sform_code"] == 0:
+            center_coordinates_world = vox_to_world_space_method_1(
+                center_coordinates, head
+            )
+        else:
+            center_coordinates_world = np.nan
+    return center_coordinates_world
+
+
+def get_center_volume(header: dict) -> ndarray:
+    """Get the voxel coordinates of the center of the data, using header information.
+
+    Parameters
+    ----------
+        header: Nifti1Header
+            Contains image metadata
+
+    Returns
+    -------
+    ndarray
+        Voxel coordinates of the center of the volume
+    """
+    import numpy as np
+
+    center_x = header["dim"][1] / 2
+    center_y = header["dim"][2] / 2
+    center_z = header["dim"][3] / 2
+    return np.array([center_x, center_y, center_z])
```

Comment:
Can be simplified in : 

```python
return np.array([_ / 2 for _ in header["dim"][1:4]])
```

Reply:
I'll change as suggested but I feel your version is more elegant computationally but not necessarily simpler.